tests/main/views/test_services.py

```diff
@@ -754,6 +755,27 @@ def test_can_change_the_supplier_id_for_a_service(self, index_service):
         # Other fields are ignored
         assert 'foo' not in data['auditEvents'][0]['data'].keys()
 
+    @mock.patch('app.main.views.services.index_service', autospec=True)
+    def test_updating_both_supplier_id_and_other_service_attributes_raises_error(self, index_service):
+        response = self._post_service_update({
+            'supplierId': 2,
+            'foo': 'bar'
+        })
+        assert response.status_code == 400
+        assert "Cannot update supplierID and other fields at the same time" in response.get_data(as_text=True)
+        assert index_service.called is False
+
+        response = self.client.get('/services/{}'.format(self.service_id))
+        data = json.loads(response.get_data())
+        # No updates made
+        assert data['services']['supplierId'] == 1
+        assert 'foo' not in data['services']
+
+        # No audit events created
+        audit_response = self.client.get('/audit-events')
+        data = json.loads(audit_response.get_data())
+        assert len(data['auditEvents']) == 0
+
```

Comment:
This matches the rest of the tests around here so it's fine, but I think we're trying to move more towards direct db checking for this kind of thing else we're testing 3 different endpoints in each of these tests which adds up to be slow if nothing else.

Reply:
Yeah, agree it's not ideal. I didn't want to go too far down a refactoring hole on this ticket though.

Reply:
Yeah no please don't.